lesson13/script.js

```diff
@@ -0,0 +1,83 @@
+function showLoadingImg(){
+    const div = document.createElement('div');
+    const loadingImg = document.createElement('img');
+    div.id = "js-loading";
+    loadingImg.src = "img/loading-circle.gif";
+    loadingImg.alt = "ローディング画像";
+    document.body.insertAdjacentElement('afterbegin',div).appendChild(loadingImg);
+}
+
+function removeLoadingImg(){
+    document.getElementById('js-loading').remove();
+}
+
+async function fetchdata(){
+    const endpoint = "https://api.json-generator.com/templates/szdgGQcOLXuk/data?access_token=hu4bc7qh9znx2m8f53mn4mz2hryvdntkavwbw8j0";
+    const response = await fetch(endpoint);
+    //responseがokではないとき
+    if(!response.ok){
+    throw new Error(`${response.status}:${response.text}`);
+    }
+    //responseがokのとき
+    const json = await response.json();
+    const data = await json.data;
+    return data;
+}
+
+function renderList(result) {
+    const fragment = document.createDocumentFragment();
+    const ul = document.getElementById('js-list');
+        for(const list of result){
+            const li = document.createElement('li');
+            const a = document.createElement('a');
+            const img = document.createElement('img');
+            a.textContent = list.text;
+            a.href = `/${list.a}`;
+            img.src = list.img;
+            img.alt = list.alt;
+            fragment.appendChild(li).appendChild(a).insertAdjacentElement('afterbegin',img);
```

Comment:
IMO
I write [課題2](https://github.com/yasu-pro/js-handson/pull/2#discussion_r626368554)
For your reference

```suggestion
　　fragment.appendChild(li).appendChild(a).prepend(img);
```